packages/components/nodes/vectorstores/Postgres/driver/PGVector.ts

```diff
@@ -0,0 +1,118 @@
+import { PoolConfig } from 'pg'
+import { getContentColumnName } from '../utils'
+
+export class PGVectorDriver extends VectorStoreDriver {
+    static CONTENT_COLUMN_NAME_DEFAULT: string = 'pageContent'
+
+    protected _postgresConnectionOptions: PoolConfig
+
+    protected async getPostgresConnectionOptions() {
+        if (!this._postgresConnectionOptions) {
+            const { user, password } = await this.getCredentials()
+            const additionalConfig = this.nodeData.inputs?.additionalConfig as string
+
+            let additionalConfiguration = {}
+
+            if (additionalConfig) {
+                try {
+                    additionalConfiguration = typeof additionalConfig === 'object' ? additionalConfig : JSON.parse(additionalConfig)
+                } catch (exception) {
+                    throw new Error('Invalid JSON in the Additional Configuration: ' + exception)
+                }
+            }
+
+            this._postgresConnectionOptions = {
+                ...additionalConfiguration,
+                host: this.getHost(),
+                port: this.getPort(),
+                user: user,
+                password: password,
+                database: this.getDatabase()
+            }
+        }
+
+        return this._postgresConnectionOptions
+    }
+
+    async getArgs(): Promise<PGVectorStoreArgs> {
+        return {
+            postgresConnectionOptions: await this.getPostgresConnectionOptions(),
+            tableName: this.getTableName(),
+            columns: {
+                contentColumnName: getContentColumnName(this.nodeData)
+            },
+            distanceStrategy: (this.nodeData.inputs?.distanceStrategy || 'cosine') as DistanceStrategy
+        }
+    }
+
+    async instanciate(metadataFilters?: any) {
+        return this.adaptInstance(await PGVectorStore.initialize(this.getEmbeddings(), await this.getArgs()), metadataFilters)
+    }
+
+    async fromDocuments(documents: Document[]) {
+        const instance = await this.instanciate()
+
+        await instance.addDocuments(documents)
+
+        return this.adaptInstance(instance)
+    }
+
+    protected async adaptInstance(instance: PGVectorStore, metadataFilters?: any): Promise<PGVectorStore> {
+        const { [FLOWISE_CHATID]: chatId, ...pgMetadataFilter } = metadataFilters
+
+        const baseSimilaritySearchVectorWithScoreFn = instance.similaritySearchVectorWithScore.bind(instance)
+
+        instance.similaritySearchVectorWithScore = async (query, k, filter) => {
+            return await baseSimilaritySearchVectorWithScoreFn(query, k, filter ?? pgMetadataFilter)
+        }
+
+        const basePoolQueryFn = instance.pool.query.bind(instance.pool)
+
+        // @ts-ignore
+        instance.pool.query = async (queryString: string, parameters: any[]) => {
+            if (!instance.client) {
+                instance.client = await instance.pool.connect()
+            }
+
+            await instance.client.connect()
```

Comment:
I have this error: 
```
Error inserting: Client has already been connected. You cannot reuse a client.
```

Reply:
hmm sorry, i implemented that because i got `Too many connections` error while using for a long period, but didn't test it since. I'll fix that.

Reply:
Done in https://github.com/FlowiseAI/Flowise/pull/3367/commits/3527383899328dc339505d01771570b42848c987